react.md

```diff
@@ -0,0 +1,242 @@
+# Trunk Club React Styleguide
+
+This guide assumes that you are writing in the coffeescript-flavor of [JSX](https://github.com/jsdf/coffee-react-transform). It also assumes that you have a cursory understanding of what JSX is and why React uses it. See the [docs](http://facebook.github.io/react/docs/jsx-in-depth.html) for more details.
+
+## Structure of React Components
+You should structure your React Components like this to ensure consistency. Whether you're using ES6 or Coffeescript or PoopyScript, it should be ordered like this
+
+1. Attributes
+  1. If you are using a mixin that requires you to specify an attribute on the Component's prototype (and not on the instance itself), do that here as well
+
+  ```Coffeescript
+  ComponentName = React.createClass
+    displayName: 'ComponentName'
+    mixins: []
+    statics: {}
+    propTypes: {}
+    listen: {} # Custom attribute needed by a mixin
+  ```
+
+2. React lifecycle hooks
+
+  ```Coffeescript
+  getInitialState: -> {}
+  componentWillMount: ->
+  componentDidMount: ->
+  componentWillReceiveProps: (nextProps) ->
+  shouldComponentUpdate: (nextProps, nextState) -> true
+  componentWillUpdate: (nextProps, nextState) ->
+  componentDidUpdate: (prevProps, prevState) ->
+  ```
+
+3. Event handlers
+
+  ```Coffeescript
+  onClickBtn: (evt) -> # on(EventType)(SourceItem)
+                       # the React docs will use `handle(EventType)`.
+                       # Same thinking, different naming
+  ```
+
+4. Helper methods for making chunks of UI
+
+  ```CoffeeScript
+  makeListOfItems: ->
+    @state.items.map (item, index) -> <li key-{index}>{item}</li>
+  ```
+
+5. `render()`
+
+  ```CoffeeScript
+  render: ->
+    <span>{@makeListOfItems()}</span>
+  ```
+
+## Code style
+1. Make sure that you are returning a single root element inside of `render()`
```

Comment:
The easiest way for us to enforce this is to have an explicit return statement at the end of `render`; otherwise, we'd have to make our own linting rule. I'll look into it :+1: